coderd/users_test.go

```diff
@@ -1721,6 +1722,83 @@ func TestSuspendedPagination(t *testing.T) {
 	require.Equal(t, expected, page.Users, "expected page")
 }
 
+func TestUserAutofillParameters(t *testing.T) {
+	t.Parallel()
+	t.Run("NotSelf", func(t *testing.T) {
+		t.Parallel()
+		client, _, api := coderdtest.NewWithAPI(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+
+		u1 := coderdtest.CreateFirstUser(t, client)
+
+		_, u2 := coderdtest.CreateAnotherUser(t, client, u1.OrganizationID)
+
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		db := api.Database
+
+		version := dbfake.TemplateVersion(t, db).Seed(database.TemplateVersion{
+			CreatedBy:      u1.UserID,
+			OrganizationID: u1.OrganizationID,
+		}).Params(database.TemplateVersionParameter{
+			Name:     "param",
+			Required: true,
+		}).Do()
+
+		_, err := client.UserAutofillParameters(
+			ctx,
+			u2.ID.String(),
+			version.Template.ID.String(),
+		)
+
+		var apiErr *codersdk.Error
+		require.ErrorAs(t, err, &apiErr)
+		require.Equal(t, http.StatusForbidden, apiErr.StatusCode())
+	})
+
+	t.Run("FindsParameters", func(t *testing.T) {
+		t.Parallel()
+		client, _, api := coderdtest.NewWithAPI(t, &coderdtest.Options{IncludeProvisionerDaemon: true})
+
+		u1 := coderdtest.CreateFirstUser(t, client)
```

Comment:
Recommend adding a separate non-admin user `u2` again, as in the above test, and testing that their authenticated client can query their parameter values.